test/serializers/serializer_for_with_namespace_test.rb

```diff
@@ -0,0 +1,85 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class SerializerForWithNamespaceTest < ActiveSupport::TestCase
+      class Book < ::Model; end
+      class Page < ::Model; end
+      class Publisher < ::Model; end
+
+      module Api
+        module V3
+          class BookSerializer < ActiveModel::Serializer
+            attributes :title, :author_name
+
+            has_many :pages
+            belongs_to :publisher
+          end
+
+          class PageSerializer < ActiveModel::Serializer
+            attributes :number, :text
+
+            belongs_to :book
+          end
+
+          class PublisherSerializer < ActiveModel::Serializer
+            attributes :name
+          end
+        end
+      end
+
+      # class ::BookSerializer < ActiveModel::Serializer
+      #   attributes :title, :author_name
+      # end
+      # test 'resource without a namespace' do
+      #   book = Book.new(title: 'A Post', author_name: 'hello')
+      #
+      #   result = ActiveModelSerializers::SerializableResource.new(book).serializable_hash
+      #
+      #   expected = { title: 'A Post', author_name: 'hello' }
+      #   assert_equal expected, result
+      # end
```

Comment:
The problem I was having with this test is that I couldn't get a root level BookSerializer to be used if a namespace wasn't present.